pydantic_ai_examples/chat_app.py

```diff
@@ -68,19 +71,51 @@ async def get_db(request: Request) -> Database:
 async def get_chat(database: Database = Depends(get_db)) -> Response:
     msgs = await database.get_messages()
     return Response(
-        b'\n'.join(MessageTypeAdapter.dump_json(m) for m in msgs),
+        b'\n'.join(ChatMessageTypeAdapter.dump_json(to_chat_message(m)) for m in msgs),
         media_type='text/plain',
     )
 
 
+class ChatMessage(TypedDict):
+    role: Literal['user', 'model']
+    timestamp: datetime
+    content: str
+
+
+def to_chat_message(m: Message) -> ChatMessage:
+    if isinstance(m, UserPrompt):
+        return {
+            'role': 'user',
+            'timestamp': m.timestamp,
+            'content': m.content,
+        }
+    elif isinstance(m, ModelResponse):
+        first_part = m.parts[0]
+        if isinstance(first_part, TextPart):
+            return {
+                'role': 'model',
+                'timestamp': m.timestamp,
+                'content': first_part.content,
+            }
+    raise UnexpectedModelBehavior(f'Unexpected message type for chat app: {m}')
+
+
+ChatMessageTypeAdapter: TypeAdapter[ChatMessage] = TypeAdapter(ChatMessage)
```

Comment:
I suppose this could be done with a like `AliasPath` thing, but this seems pretty simple, and allows us to raise an error cleanly.

Reply:
I'm not sure `ChatMessageTypeAdapter` is necessary at all, might be cleaner to just use `json.dumps`, and it's clearr.